File: 242.valid-anagram/solution.kt
class Solution {
    fun isAnagram(s: String, t: String): Boolean {
        if (s.length != t.length) return false

        var counts = ArrayList<Int>()
        for (i in 0 .. 25) {
            counts.add(0)
        }

        val aOrd = 'a'.toInt()
        for (i in 0 until s.length) {
            counts[s[i].toInt() - aOrd]++
        }
        for (i in 0 until t.length) {
            counts[t[i].toInt() - aOrd]--
        }
        for (i in 0..25) {
            if (counts[i] != 0) return false
        }
        return true
    }
}

File: 242.valid-anagram/solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if (len(s) != len(t)):
            return False

        counts = [0] * 26
        aOrd = ord("a")
        for char in s:
            idx = ord(char) - aOrd
            counts[idx] += 1
        for char in t:
            idx = ord(char) - aOrd
            counts[idx] -= 1

        for i in range(26):
            if (counts[i] != 0):
                return False
        return True
